### PR TITLE
update documentation, refactoring class RPClient, increase node version

### PR DESCRIPTION
**Pull request by DzmitryRamaniuk**

### CODE REVIEW

**Review by lexecon — APPROVED** [review]

**Review by lexecon — COMMENTED** [review]
I checked only the syntax of the code

File: lib/client.js [hunk 1]
```diff
@@ -32,89 +30,104 @@ class RPClient {
...
-                data: _self.buildMultiPartStream(saveLogRQ, file, boundary),
+            rest.postJson(url, {
+                data: this.buildMultiPartStream(saveLogRQ, file, boundary),
                 headers: {
-                    "Authorization": 'bearer ' + _self.token,
+                    "Authorization": 'bearer ' + this.token,
                     "Content-Type": 'multipart/form-data; boundary=' + boundary
                 }
-            }).on('complete', function (data, r) {
-                resolve(data);
```
Comment (dshlychkou) [inline lib/client.js:106]: Could please describe the purpose of this change?

File: lib/client.js [hunk 2]
```diff
@@ -32,89 +30,104 @@ class RPClient {
...
-                .on("fail", function (data) {
-                    reject(data);
-                })
-                .on("error", function (data) {
-                    reject(data);
+        let promise = new Promise((resolve, reject) => {
+            if (method === "POST") {
+            return rest.postJson(url, rq, options)
+                .on('complete', result => {
+                    if (result instanceof Error) {
```
Comment (dshlychkou) [inline lib/client.js]: Could you please describe the purpose of this change?

File: lib/client.js [hunk 3]
```diff
@@ -32,89 +30,104 @@ class RPClient {
...
-                    "Authorization": 'bearer ' + _self.token,
+                    "Authorization": 'bearer ' + this.token,
                     "Content-Type": 'multipart/form-data; boundary=' + boundary
                 }
-            }).on('complete', function (data, r) {
-                resolve(data);
-            }).on('fail', (data, res) => {
-                console.log(res.statusCode, data)
+            }).on('complete', result => {
+                    if (result instanceof Error) {
```
Comment (dshlychkou) [inline lib/client.js]: Could you please provide the purpose of this change?

File: lib/client.js [hunk 4]
```diff
@@ -32,89 +30,104 @@ class RPClient {
...
+                    } else {
+                        resolve(result);
+                    }
                 });
-        };
-        return new Promise(response);
+            } else if (method === "PUT") {
+                return rest.putJson(url, rq, options)
+                    .on('complete', result => {
+                        if (result instanceof Error) {
```
Comment (dshlychkou) [inline lib/client.js]: Could you please describe the purpose of this change?

**Review by avarabyeu — CHANGES_REQUESTED** [review]
Please, answer on Denis's questions

**Pull request closed without merging**